scene/resources/mesh.h

```diff
@@ -42,6 +42,7 @@ class Mesh : public Resource {
 	GDCLASS(Mesh, Resource);
 
 	mutable Ref<TriangleMesh> triangle_mesh; //cached
+	mutable Vector<Ref<TriangleMesh>> surface_triangle_meshes; //cached
```

Comment:
I think rather than doing this, it should be simpler to modify TriangleMesh to contain an index of the surface ID as an optional second argument Vector<int32_t> to triangle_mesh->create()

Reply:
This will also ensure that ray collision is correct.